src/components/Department.tsx

```diff
@@ -0,0 +1,43 @@
+import Image from "next/image";
+import blogIcon from "../../public/blog.svg";
+
+const Department = ({ from }: { from: string }) => {
+	let txtSize;
+	let height;
+	let width;
+	let border;
+    let bgColor
+    let ghost
+	if (from === "nav") {
+		txtSize = "xl ml-4";
+		height = 50;
+		width = 40;
+		border = "";
+        bgColor = ""
+        ghost = ""
+	} else if (from === "publisher") {
+		txtSize = "l mt-3";
+		height = 30;
+		width = 20;
+		border = "border-4 border-green-600 h-6";
+        bgColor = "bg-base-300" 
+        ghost = "btn-ghost"
+	}
```

Comment:
❌ 
this way of setting vars is so not DRY (Don't Repeat Yourself), also it does not follow the OCP (Open Close Principle) principle from the SOLID principle by uncle boob, research this to learn more about it,

Reply:
you could create a function that will return an object with the right props instead